src/soap/soap-client-callback.ads

```diff
@@ -0,0 +1,34 @@
+with SOAP.Message.Payload;
+with SOAP.Message.Response;
+
+package SOAP.Client.Callback is
+
+   --  Callback routine types
+   type Pre_Call_CB is not null access
+      procedure (Connection : AWS.Client.HTTP_Connection;
+                 SOAPAction : String;
```

Comment:
This SOAPAction is part of the Request parameter. Not sure it is needed in the CB, one can get it with

   SOAPAction : constant String := Status.SOAPAction (Request);

Reply:
Probably better to do it like that then.